sim/battle.ts

```diff
@@ -3196,10 +3357,100 @@ export class Battle extends Dex.ModdedDex {
 		throw new Error(`The tryMoveHit function needs to be implemented in scripts.js or the battle format.`);
 	}
 
+	trySpreadMoveHit(target: Pokemon[], pokemon: Pokemon, move: Move): boolean {
+		throw new Error(`The trySpreadMoveHit function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	tryImmunityEvent(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): boolean[] {
+		throw new Error(`The tryImmunityEvent function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	typeImmunity(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): boolean[] {
+		throw new Error(`The typeImmunity function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	tryHitEvent(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): (boolean | '')[] {
+		throw new Error(`The tryHitEvent function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	powderImmunity(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): boolean[] {
+		throw new Error(`The powderImmunity function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	pranksterImmunity(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): boolean[] {
+		throw new Error(`The pranksterImmunity function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	accuracy(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): boolean[] {
+		throw new Error(`The accuracy function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	breakProtect(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): undefined {
+		throw new Error(`The breakProtect function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	stealBoosts(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): undefined {
+		throw new Error(`The stealBoosts function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	afterMoveSecondaryEvent(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): undefined {
+		throw new Error(`The afterMoveSecondaryEvent function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	moveHitLoop(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): (number | undefined | boolean)[] {
+		throw new Error(`The moveHitLoop function needs to be implemented in scripts.js or the battle format.`);
+	}
+
+	spreadMoveHit(
+		targets: (Pokemon | false | null)[], pokemon: Pokemon, move: ActiveMove, moveData?: ActiveMove,
```

Comment:
maybe use `type` aliases for ` (Pokemon | false | null)[]` and `(number | undefined | boolean)[]` if you can figure out a name for them?